go-controller/pkg/ovn/controller/services/load_balancer.go

```diff
@@ -343,10 +359,21 @@ func buildPerNodeLBs(service *v1.Service, configs []lbConfig, nodes []nodeInfo)
 							Targets: targetsETP,
 						})
 					}
-					switchRules = append(switchRules, ovnlb.LBRule{
-						Source:  ovnlb.Addr{IP: vip, Port: config.inport},
-						Targets: targets,
-					})
+					if config.internalTrafficLocal && config.svcType == "ClusterIP" && util.IsClusterIP(vip) { // ITP only applicable to CIP
```

Comment:
I'm not sure I agree with `config.svcType == "ClusterIP"` - all that should matter is whether or not the ClusterIP is set.

Reply:
haha actually that's a good point, I just thought the config.svcType == "ClusterIP" was an extra filter since this won't match on the nodePorts and ETP cases LBConfigs.
Do you think I should probably just remove that match?

Reply:
Hmm ClusterIPs will be set for all those different type of svc but since ITP is only for intra cluster traffic I thought we should limit it to svcType ClusterIP as well ?

Reply:
ClusterIPs should have the same behavior, regardless of service Type. So I think we should not look at service Type unless really necessary.

Reply:
@tssurya can you just get rid of config.svcType and in the above NodePort check if vip !isClusterIP?

Reply:
yes let me do this.